### PR TITLE
Use audit policy and splunk secret from resource references.

### PR DESCRIPTION
**Pull request by Gerrit91**
- The audit policy can already be defined in the shoot spec, no need to have an own field for this
- Use the resource references in the shoot spec to provide the splunk secret, this prevents any direct secrets exposed in the shoot resource
- Do not deploy audittailer to shoot if cluster forwarding is not enabled
- Update README

### CODE REVIEW

**Review by vknabel — COMMENTED** [review]
Due to the breaking changes, does it make sense to write a migration document that we can follow during upgrade?

File: example/shoot.yaml [hunk 1]
```diff
@@ -17,41 +18,26 @@ spec:
       apiVersion: audit.metal.extensions.gardener.cloud/v1alpha1
       kind: AuditConfig
       webhookMode: blocking
+      # persistence:
+      #   size: 10Gi
```
[the remark this answers is not in the document]
Comment (Gerrit91) [reply example/shoot.yaml:22]: These are intentional as this is how the size of the persistent volume can be specified. In the Gardener repo this is also done in the examples.

**Comment by Gerrit91** [comment]
Breaking changes do not matter now as this extension is only a couple of days old and there was not a single release yet. :D

**Review by vknabel — APPROVED** [review]
I couldn't do much testing, but the code looks good. :+1:

**Pull request merged**